### ISSUE TITLE
logrotate: Crashes

### ISSUE THREAD

**Issue by neilmayhew** [NixOS/nixpkgs#345602]
### Describe the bug

In `configuration.nix`:

```nix
services.logrotate.enable = true;
```

After booting:

```shellsession
$ systemctl status logrotate.service 
× logrotate.service - Logrotate Service
     Loaded: loaded (/etc/systemd/system/logrotate.service; linked; preset: enabled)
     Active: failed (Result: core-dump) since Mon 2024-09-30 15:00:00 MDT; 45min ago
 Invocation: 122606e090584167b1b2b00d52edb852
TriggeredBy: ● logrotate.timer
       Docs: man:logrotate(8)
             man:logrotate(5)
    Process: 4249 ExecStart=/nix/store/73h2hniwk0zb3k9zjmvlbm80944kpvpw-logrotate-3.22.0/bin/logrotate /nix/store/fs7ipiag4lpl87zg9d33n2cmzpf4hc93-logrotate.conf (code=dumped, signal=SYS)
   Main PID: 4249 (code=dumped, signal=SYS)
         IP: 0B in, 0B out
   Mem peak: 2.4M
        CPU: 19ms

Sep 30 15:00:00 <hostname> systemd[1]: Starting Logrotate Service...
Sep 30 15:00:00 <hostname> systemd[1]: logrotate.service: Main process exited, code=dumped, status=31/SYS
Sep 30 15:00:00 <hostname> systemd[1]: logrotate.service: Failed with result 'core-dump'.
Sep 30 15:00:00 <hostname> systemd[1]: Failed to start Logrotate Service.
```

### Steps To Reproduce
Enable `logrotate` in `configuration.nix` and `nixos-rebuild switch` or `nixos-rebuild boot` followed by a restart.

### Expected behavior
`logrotate` doesn't crash.

### Additional context
```shellsession
$ sudo coredumpctl debug
...
Program terminated with signal SIGSYS, Bad system call.
...
(gdb) bt
#0  0x00007fa0a6d879fd in seteuid () from /nix/store/3dyw8dzj9ab4m8hv5dpyx7zii8d0w6fi-glibc-2.39-52/lib/libc.so.6
#1  0x000000000040d781 in switch_user ()
#2  0x00000000004052ce in main ()
```

In the `logrotate.conf`:
```shellsession
$ tail -n11 /nix/store/fs7ipiag4lpl87zg9d33n2cmzpf4hc93-logrotate.conf
"/var/log/httpd/*.log" {
  compress
  delaycompress
  daily
  postrotate
    systemctl reload httpd.service > /dev/null 2>/dev/null || true
  endscript
  rotate 28
  sharedscripts
  su wwwrun wwwrun
}
```

In the unit file:
```shellsession
$ systemctl cat logrotate.service | grep -i '[gu]id'
CapabilityBoundingSet=CAP_SETUID
CapabilityBoundingSet=CAP_SETGID
RestrictSUIDSGID=false
SystemCallFilter=@system-service @setuid
```

Changes since the last working system profile:
```shellsession
$ diff /nix/var/nix/profiles/system-44{7,9}-link/etc/systemd/system/logrotate.service 
9a10,11
> CapabilityBoundingSet=CAP_DAC_OVERRIDE
> CapabilityBoundingSet=CAP_SETUID
35c37
< RestrictSUIDSGID=true
---
> RestrictSUIDSGID=false
37c39
< SystemCallFilter=@system-service
---
> SystemCallFilter=@system-service @setuid
```

The `logrotate` derivation hasn't changed recently, so this is presumably a systemwide change to unit files that happens not to work for `logrotate`.

### Notify maintainers

@tobim 

### Metadata
```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.6.52, NixOS, 24.11 (Vicuna), 24.11pre687049.06cf0e1da420`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.5`
 - channels($USER): `"nixpkgs"`
 - channels(root): `"nixos"`
 - nixpkgs: `$HOME/.nix-defexpr/channels/nixpkgs`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by exi** [comment]
Can confirm, broken for me as well.

**Comment by h7x4** [comment]
cc @mweinelt

**Comment by mweinelt** [comment]
Likely fixed in #345275 and #345487

**Comment by neilmayhew** [comment]
Thank you, @mweinelt. That seems to have fixed it.
